components/sidebar/sidebar_channel_button_or_link/sidebar_channel_button_or_link.jsx

```diff
@@ -93,6 +94,7 @@ export default class SidebarChannelButtonOrLink extends React.PureComponent {
         } else {
             element = (
                 <Link
+                    id={`sidebarItem_${this.props.channelName}`}
```

Comment:
Instead of adding an extra prop, would it be possible to re-use the channel ID here? I'm not particularly familiar with how you'd get that data in Cypress, but I'd prefer to use the existing data instead of passing the channel name around just for this. Maybe @saturninoabril would have some ideas since he's more familiar with the testing enviornment

Reply:
Actually, I've [suggested](https://github.com/mattermost/mattermost-webapp/pull/2251#discussion_r245867952) that particular change.  I've looked at using channel ID but it will require complicated test logic.  Using channel name is straightforward for testing.  Besides, this dynamic ID will be used frequently in testing the channel sidebar.

Reply:
I guess that's fine here since chances are that extra prop won't change often enough to affect performance, although I think there might be other places like the post list where we'll need to come up with something more generic as we can't rely on something hardcoded like a channel name there. It would be interesting for those if we could somehow pass that part of the redux state out to Cypress somehow or have the IDs be constant somehow.

Reply:
re: redux state, interesting.  I've looked into it and it's indeed possible.  I'll attempt to experiment with that.  Thanks!